### PR TITLE
[AKS] aks enable-addons: Add `--enable-syslog` parameter to monitoring addon

### PR DESCRIPTION
**Pull request by pfrcks**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 1]
```diff
@@ -270,6 +271,9 @@
   - name: --enable-msi-auth-for-monitoring
     type: bool
     short-summary: Enable Managed Identity Auth for Monitoring addon.
+  - name: --enable-syslog
+    type: bool
+    short-summary: Enable syslog data collection for AKS clusters
```
Comment (ganga1980) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:276]: nit: you can say "Enable syslog data collection for Monitoring Addon"

File: src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py [hunk 2]
```diff
@@ -435,11 +437,94 @@ def ensure_container_insights_for_monitoring(
                     },
                 }
             )
+
```
Comment (ganga1980) [inline src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py:440]: @pfrcks, changes looks good to me. just want to sure functionality works as expected. Did you verify enabling monitoring addon with syslog on existing and also as part of the cluster creation?

Comment (pfrcks) [reply src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py:480]: yes verified with both existing and new

**Review by ganga1980 — APPROVED** [review]
lgtm

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -6924,7 +6924,21 @@ def test_aks_create_with_monitoring_aad_auth_uai(self, resource_group, resource_
         aks_name = self.create_random_name('cliakstest', 16)
         self.create_new_cluster_with_monitoring_aad_auth(resource_group, resource_group_location, aks_name, user_assigned_identity=True)
 
-    def create_new_cluster_with_monitoring_aad_auth(self, resource_group, resource_group_location, aks_name, user_assigned_identity=False):
+    @live_only()
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6927]: May I ask why these tests are marked as live only?

Comment (pfrcks) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6927]: these are based off of the two tests above which were also marked as @live_only()

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6927]: @pfrcks If these tests are marked as `@live_only()`, they cannot be rerun by our PR CI pipeline. 
It is suggested to investigate the reason why these tests are marked as `@live_only()`, and then solve those `@live_only()` tests in a separate new PR, do you think it is OK?

Comment (pfrcks) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6927]: sure, I will add a task to investigate this in our Dec sprint

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**